tmt/steps/provision/__init__.py

```diff
@@ -485,6 +489,38 @@ def stop(self):
         """
         self.debug(f"Doing nothing to stop guest '{self.guest}'.")
 
+    def reboot(self, hard=False, machine_name=None):
```

Comment:
How is the `machine_name` supposed to be working?

Reply:
When you want to reboot machine from cli as like in previous version:
tmt -h connect reboot --name 123.456.789.123

Reply:
Well, the parameter is never used in the method. That's why I'm asking. The other thing is, we currently support just a single guest for each plan. So there's no point in selecting the guest. If there are multiple plans in the run, you can select the desired plan instead:

    tmt run plan --name selected

In the (near) future we'll be adding spec and implementation for multiple guests (including the guest naming). What about implementing the machine selection for reboot only when multiple guests in a single plan are implemented?

Reply:
Ok. So I can remove this for now.

Reply:
Removed then solved :).